osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -310,6 +313,14 @@ private List<BeatmapInfo> createBeatmapDifficulties(ArchiveReader reader)
             return beatmapInfos;
         }
 
+        private double calculateLength(IBeatmap b)
+        {
+            var lastObject = b.HitObjects.LastOrDefault();
```

Comment:
```csharp

if (!b.HitObjects.Any())
    return 9;

var lastObject = b.HitObjects.Last();

double endTime = (lastObject as IHasEndTime)?.EndTime ?? lastObject.StartTim;
double startTime = b.HitObjects.First().StartTime;

return endTime - startTime;
```

should read better

Reply:
you meant 0 for `!b.HitObjects.Any()` right?